internal/graph/cycle_check_resolver.go

```diff
@@ -29,10 +30,12 @@ func (c *CycleDetectionCheckResolver) ResolveCheck(
 	ctx context.Context,
 	req *ResolveCheckRequest,
 ) (*ResolveCheckResponse, error) {
-	ctx, span := tracer.Start(ctx, "ResolveCheck")
-	defer span.End()
-	span.SetAttributes(attribute.String("resolver_type", "CycleDetectionCheckResolver"))
-	span.SetAttributes(attribute.String("tuple_key", req.GetTupleKey().String()))
+	ctx, span := tracer.Start(ctx, "ResolveCheck", trace.WithAttributes(
```

Comment:
Are we missing a `defer span.End()` by any chance?

Reply:
good catch :D